ui/marketplace/MarketplaceAppTopBar.tsx

```diff
@@ -0,0 +1,106 @@
+import { chakra, Flex, Tooltip, Skeleton, Box } from '@chakra-ui/react';
+import React from 'react';
+
+import type { MarketplaceAppOverview } from 'types/client/marketplace';
+
+import { route } from 'nextjs-routes';
+
+import { useAppContext } from 'lib/contexts/app';
+import type { IconName } from 'ui/shared/IconSvg';
+import IconSvg from 'ui/shared/IconSvg';
+import LinkExternal from 'ui/shared/LinkExternal';
+import LinkInternal from 'ui/shared/LinkInternal';
+
+import MarketplaceAppInfo from './MarketplaceAppInfo';
+
+type Props = {
+  isWalletConnected: boolean;
+  data: MarketplaceAppOverview | undefined;
+  isPending: boolean;
+}
+
+const MarketplaceAppTopBar = ({ data, isPending, isWalletConnected }: Props) => {
+  const appProps = useAppContext();
+
+  const goBackUrl = React.useMemo(() => {
+    if (appProps.referrer && appProps.referrer.includes('/apps') && !appProps.referrer.includes('/apps/')) {
+      return appProps.referrer;
+    }
+    return route({ pathname: '/apps' });
+  }, [ appProps.referrer ]);
+
+  const message = React.useMemo(() => {
+    let icon: IconName = 'wallet';
+    let iconColor = 'blackAlpha.800';
+    let bgColor = 'orange.100';
+    let text = 'Connect your wallet to Blockscout for full-featured access';
+
+    if (isWalletConnected && data?.internalWallet) {
+      icon = 'integration/full';
+      iconColor = 'green.600';
+      bgColor = 'green.100';
+      text = 'Your wallet is connected with Blockscout';
+    } else if (isWalletConnected) {
+      icon = 'integration/partial';
+      text = 'Connect your wallet in the app below';
+    }
+
+    return { icon, iconColor, bgColor, text };
+  }, [ isWalletConnected, data?.internalWallet ]);
+
+  if (isPending) {
+    return (
+      <Flex alignItems="center" mb={ 2 } gap={ 2 }>
+        <Skeleton w="25px" h="32px" borderRadius="base"/>
+        <Skeleton w="300px" h="32px" borderRadius="base"/>
+        <Skeleton w="75px" h="32px" borderRadius="base"/>
+        <Skeleton w="150px" h="32px" borderRadius="base"/>
+      </Flex>
+    );
+  }
```

Comment:
The preferred approach is to enclose each element in the `Skeleton` component. This ensures that the loading layout remains consistent with the actual layout.

Note, that some of our components already implement the loading state. So you just have to pass the prop to them.